prime-dotnet-webapi/Controllers/HealthAuthoritySitesController.cs

```diff
@@ -264,12 +272,50 @@ public async Task<ActionResult> HealthAuthoritySiteSubmission(int healthAuthorit
             {
                 return BadRequest("Action could not be performed.");
             }
+            var authorizedUser = await _authorizedUserService.GetAuthorizedUserForUsernameAsync(User.GetPrimeUsername());
 
-            await _healthAuthoritySiteService.UpdateSiteAsync(siteId, updateModel);
+            await _healthAuthoritySiteService.UpdateSiteAsync(siteId, updateModel, authorizedUser.Id);
             await _healthAuthoritySiteService.SiteSubmissionAsync(siteId);
+            await _businessEventService.CreateSiteEventAsync(siteId, User.GetPrimeUserId(), "Health Authority Site has been updated and submitted");
 
             await _emailService.SendHealthAuthoritySiteRegistrationSubmissionAsync(siteId);
-            await _businessEventService.CreateSiteEmailEventAsync(siteId, "Notified of health authority site registration submission");
+            await _businessEventService.CreateSiteEmailEventAsync(siteId, User.GetPrimeUserId(), "Notified of health authority site registration submission");
+
+            return NoContent();
+        }
+
+
+        // POST: api/health-authorities/5/sites/transfer/from/10/to/11
+        /// <summary>
+        /// Transfer site from one authorized user to another within a health authority.
+        /// </summary>
+        /// <param name="healthAuthorityId"></param>
+        /// <param name="currentAuthorizedUserId"></param>
+        /// <param name="newAuthorizedUserId"></param>
+        [HttpPost("transfer/from/{currentAuthorizedUserId}/to/{newAuthorizedUserId}", Name = nameof(HealthAuthoritySiteTransfer))]
+        [Authorize(Roles = Roles.PrimeSuperAdmin)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(typeof(ApiMessageResponse), StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        public async Task<ActionResult> HealthAuthoritySiteTransfer(int healthAuthorityId, int currentAuthorizedUserId, int newAuthorizedUserId)
+        {
+            var authorizedUsers = await _healthAuthorityService.GetAuthorizedUsersAsync(healthAuthorityId);
+
+            if (authorizedUsers.Any() &&
+                authorizedUsers.Any(au => au.Id == currentAuthorizedUserId) &&
+                authorizedUsers.Any(au => au.Id == newAuthorizedUserId))
+            {
+                var siteIds = await _healthAuthoritySiteService.TransferAuthorizedUserAsync(currentAuthorizedUserId, newAuthorizedUserId);
+                foreach (int siteId in siteIds)
+                {
+                    await _businessEventService.CreateSiteEventAsync(siteId, "Health Authorized Site is transfered to other authorized user");
```

Comment:
I think that looking at event log later, it would be more useful to name the users transferred from and transferred to, as part of the message